### PR TITLE
Task/API 383 Add Medication Order API Model

### PR DESCRIPTION
**Pull request by esubocz-va**
API model for Medication Order resource has been added to argonaut-api project.
Related story:
https://vasdvp.atlassian.net/browse/API-383

### CODE REVIEW

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    String id;
+
+    @Valid List<Extension> extension;
+    @Valid List<Extension> modifierExtension;
+
+    @Valid CodeableConcept medicationCodeableConcept;
+    @Valid Reference medicationReference;
+    @Valid Period validityPeriod;
+
+    @Min(0)
```
Comment (eclendenning-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java]: Should be @Min(1)
positiveInteger defined as Any positive integer (e.g. > 0) in spec.

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+  @Valid List<SimpleResource> contained;
+  @Valid List<Extension> extension;
+  @Valid List<Extension> modifierExtension;
+  @Valid List<Identifier> identifier;
+
+  @NotBlank
+  @Pattern(regexp = Fhir.DATETIME)
+  String dateWritten;
+
+  @NotNull @Valid Status status;
```
Comment (eclendenning-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java]: @Valid not needed here

**Comment by esubocz-va** [comment]
All suggested changes have been addressed, good catches! @eclendenning-va

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/datatypes/Timing.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    mo,
+    a
+  }
+
+  @Data
+  @Builder
+  @NoArgsConstructor(access = AccessLevel.PRIVATE)
+  @AllArgsConstructor
+  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @ZeroOrOneOfs(@ZeroOrOneOf(fields = {"boundsQuantity", "boundsRange", "boundsPeriod"}))
```
Comment (bschofield-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/datatypes/Timing.java]: If only one group, you can omit the outer wrapper annotation, e.g.
```
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@ZeroOrOneOf(fields = {"boundsQuantity", "boundsRange", "boundsPeriod"})
public static class Repeat implements Element {
```

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+      super(resourceType, id, meta, implicitRules, language, type, total, link, entry, signature);
+    }
+  }
+
+  @Data
+  @Builder
+  @NoArgsConstructor(access = AccessLevel.PRIVATE)
+  @AllArgsConstructor
+  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @ZeroOrOneOfs({
```
Comment (bschofield-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java]: Don't need `@ZeroOrOneOfs` wrapper annotation

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+@Data
+@Builder
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor
+@JsonAutoDetect(fieldVisibility = Visibility.ANY)
+@Schema(
+  description =
+      "http://www.fhir.org/guides/argonaut/r2/StructureDefinition-argo-medicationorder.html"
+)
+@ZeroOrOneOfs({
```
Comment (bschofield-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java]: Don't need `@ZeroOrOneOfs` or `ExactlyOneOfs` wrapper annotations

**Comment by esubocz-va** [comment]
I fix.

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+  @Valid List<Extension> extension;
+  @Valid List<Extension> modifierExtension;
+  @Valid List<Identifier> identifier;
+
+  @NotBlank
+  @Pattern(regexp = Fhir.DATETIME)
+  String dateWritten;
+
+  @NotNull
+  @Pattern(regexp = Fhir.CODE)
```
Comment (bschofield-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/MedicationOrder.java]: Should be an enum

**Pull request merged**